shell/golangci-lint.sh

```diff
@@ -15,10 +15,28 @@ if [[ -z $workspaceFolder ]]; then
 fi
 
 # Enable only fast linters, and always use the correct config.
-args=("--config=${workspaceFolder}/scripts/golangci.yml" "$@" "--fast" "--allow-parallel-runners")
+args=("--config=${workspaceFolder}/scripts/golangci.yml" "$@"  "-v" "--fast" "--allow-parallel-runners")
 
-# trade memory for CPU when running the linters
-export GOGC=20
+
+asdf_devbase_exec golangci-lint --version
+
+# If we're on a system with free, set GOMEMLIMIT to a value that's less
+# than the max amount of RAM on the system. This helps ensure that we
+# don't go over the memory limit and get OOM killed. This is mostly
+# important for CI systems.
+if command -v free &>/dev/null; then
+  mem="$(free -m | awk '/^Mem:/{print $2}')"
+
+  # Use mem as the memory target and ensure that we have 2GB of room.
+  export GOMEMLIMIT="$((mem - 2048))MiB"
```

Comment:
thought: seems like there should be a check to see if `free` returns a value > 2048.